packages/tupaia-web/src/features/EntitySearch/EntitySearch.tsx

```diff
@@ -58,16 +58,22 @@ export const EntitySearch = () => {
   const [isOpen, setIsOpen] = useState(false);
   const onClose = () => {
     setIsOpen(false);
+    gaEvent('Search', 'Toggle Expand');
     setSearchValue('');
   };
-  useEffect(() => {
-    gaEvent('Search', 'Change Search');
-  }, [setSearchValue]);
-  useEffect(() => {
-    if (isOpen === true) {
-      gaEvent('Search', 'Open Search Bar');
+
+  const toggleExpandedSearchBar = () => {
+    setIsOpen(!isOpen);
+    gaEvent('Search', 'Toggle Expand');
+    if (isOpen === false) {
```

Comment:
Just wondering if we should have like `const updatedOpenValue =. !isOpen` and then refer to that when checking whether to set search value as `''`. Just because there is sometimes a split second before it's updated